Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,212 @@
+
+    // fixed keys
+    static let formatKey = CodingKeys(key: "NSStringLocalizedFormatKey")
+    static let variableWidthRules = CodingKeys(key: "NSStringVariableWidthRuleType")
+  }
+
+  init(from coder: Decoder) throws {
+    let container = try coder.container(keyedBy: CodingKeys.self)
+    let formatKey = try container.decodeIfPresent(String.self, forKey: .formatKey)
+    let variableWidthRules = try container.decodeIfPresent([String: String].self, forKey: .variableWidthRules)
+
+    // We either have a formatKey OR we have a variableWidthRule.
+    switch (formatKey, variableWidthRules) {
+    case (.none, .none), (.some, .some):
+      throw Strings.ParserError.invalidFormat
+    case let (.some(formatKey), .none):
+      let variables = try StringsDict.decodeVariableRules(in: container, formatKey: formatKey)
+      self = .pluralEntry(PluralEntry(formatKey: formatKey, variables: variables))
+    case let (.none, .some(variableWidthRules)):
+      self = .variableWidthEntry(VariableWidthEntry(rules: variableWidthRules))
+    }
+  }
+
+  private static func decodeVariableRules(
+    in container: KeyedDecodingContainer<StringsDict.CodingKeys>,
+    formatKey: String
+  ) throws -> [PluralEntry.Variable] {
+    guard let variableKeyTuples = StringsDict.variableKeysFromFormatKey(formatKey) else { return [] }
+    let sortedVariableKeys = variableKeyTuples
+      .sorted { lhs, rhs in
+        // Sort by positional argument if present, otherwise by occurrence in the format key
+        switch (lhs.positionalArgument, rhs.positionalArgument, lhs.range.lowerBound, rhs.range.lowerBound) {
+        case let (.none, .none, lhs, rhs):
+          return lhs < rhs
+        case (.some, .none, _, _):
+          return true
+        case (.none, .some, _, _):
+          return false
+        case let (.some(lhs), .some(rhs), _, _):
+          return lhs < rhs
+        }
+      }
+      .map { $0.key }
+
+    return try sortedVariableKeys.reduce(into: [PluralEntry.Variable]()) { variables, variableKey in
+      let variableRule = try container.decode(PluralEntry.VariableRule.self, forKey: CodingKeys(key: variableKey))
+      variables.append(PluralEntry.Variable(key: variableKey, rule: variableRule))
+    }
+  }
+}
+
+// - MARK: Helpers
+
+extension StringsDict {
+  typealias VariableKeyResult = (key: String, range: Range<String.Index>, positionalArgument: Int?)
+
+  /// Parses variable keys and their ranges from a formatKey.
+  /// Every variable key that is contained within the format string is preceded
+  /// by the %#@ characters or by the positional variants, e.g. %1$#@, and followed by the @ character.
+  ///
+  /// - Parameter formatKey: The formatKey from which the variable keys should be parsed.
+  /// - Returns: An array of discovered variable keys and their range within the `formatKey`.
+  // swiftlint:disable:next discouraged_optional_collection
+  private static func variableKeysFromFormatKey(_ formatKey: String) -> [VariableKeyResult]? {
```

Comment:
nit: rename this method to `variableNamesFromFormatKey`?

Reply:
I originally chose this name, because the name of the variable is used as key, but you are right, the actual thing that we parse is the _name_ and the _name_ is only used as _key_ in the `.stringsdict`.